2018-11-02  Gajan Rathnam Nagaraj  <gajan@lnxsrv09.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	There is an integer overflow issue for edge case inputs to this function. We 
  change this so that we do a direct comparison of the inputs and return 1, 0, 
  or -1.

--- lib/timespec.h 2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h  2018-11-02 10:56:08.649247000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec < b.tv_sec){
+    return 1;
+  }
+  else if (a.tv_sec < b.tv_sec){
+    return -1;
+  }
+  else {
+    return 0;
+  }
 }

 # if ! HAVE_DECL_NANOSLEEP

